flycheck.el

```diff
@@ -1501,6 +1501,13 @@ FILE-NAME is nil, return `default-directory'."
           (file-name-directory file-name)
         (expand-file-name default-directory)))))
 
+(defun flycheck-line-column-to-position (line column)
```

Comment:
Is there a function that does the reverse of this?  [My eglot checker](https://gist.github.com/eyal0/8f58d4209dc8cdb5d9c474b61c49b2dc) already provides the start and end character position.

Reply:
I think the best would be to add a custom constructor, so that your code just gets magically faster once we support both representations without having to compute line and column numbers.  I'll do that in a separate PR, or do you want to do it? The idea is to go through the conversion for now, and at a later point we can change the constructor.

Reply:
I don't feel confident enough at elisp to do this.  Maybe I will try after this PR, if I can get assistance from you in reviewing the code.

Reply:
Does this help? https://github.com/purcell/package-lint/blob/cf2ce871549bb816ecb7a9930bdfb77525468f5e/package-lint.el#L201-L207

Reply:
`current-column` looks suspicious in that snippet, since we consider the column number a number of chars in other parts of the code.

Reply:
@eyal0 always happy to help

Reply:
Just for the record - the LSP protocol returns line + column not point.

Reply:
> `current-column` looks suspicious in that snippet, since we consider the column number a number of chars in other parts of the code.

Ah yes, looks like `package-lint` uses zero-based columns for easy compatibility with internal Emacs functions, and then `flycheck-package` uses `flycheck-increment-error-columns` to change them to `flycheck`'s 1-based columns (which presumably originated from that being the convention in most compiler/linter output).

Reply:
@purcell it's not enough to add/subtract one.  `(current-column)` is giving the visual column.  For example, after a single tab by itself on a line, the number might be `8`, even though there is just one character.  Probably it should return 1 or maybe 2.

Reply:
UGH. I guess then `(- (point) (line-beginning-position))`....

Reply:
@purcell still wrong, sorry. You need to add one because flycheck assumes that columns begin at one. `(1+ ...)` 😜